src/libraries/Common/src/Interop/Unix/System.Native/Interop.MkDir.cs

```diff
@@ -3,12 +3,21 @@
 
 using System;
 using System.Runtime.InteropServices;
+using System.Text;
 
 internal static partial class Interop
 {
     internal static partial class Sys
     {
-        [GeneratedDllImport(Libraries.SystemNative, EntryPoint = "SystemNative_MkDir", CharSet = CharSet.Ansi, SetLastError = true)]
-        internal static partial int MkDir(string path, int mode);
+        [GeneratedDllImport(Libraries.SystemNative, EntryPoint = "SystemNative_MkDir", SetLastError = true)]
+        private static partial int MkDir(ref byte path, int mode);
```

Comment:
Doesn't GeneratedDllImport support spans directly?  Or maybe that hasn't been implemented yet? (@AaronRobinsonMSFT ?)

Reply:
Yes, we added testing support for `Span<T>`. It requires specifying the [`SpanMarshaller<T>`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime.InteropServices/tests/Ancillary.Interop/SpanMarshallers.cs) via a `MarshalUsing` attribute though. It is in the `Ancillary.Interop` assembly which is for our proposed public APIs the platform will provide.

More details on the marshaller are [here](https://github.com/dotnet/runtime/blob/main/docs/design/libraries/DllImportGenerator/SpanMarshallers.md).

/cc @jkoritzinsky @elinor-fung

Reply:
This is what I got when I've tried to use `ROS<char>`:

```log
error DLLIMPORTGEN001: The type 'System.ReadOnlySpan<char>' is not supported by source-generated P/Invokes. The generated source will not handle marshalling of parameter 'path'. [/home/adam/projects/runtime/src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj]
```

It would be great if the error mentioned `SpanMarshaller<T>`

Reply:
@AaronRobinsonMSFT do I understand correctly that [`SpanMarshaller<T>`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime.InteropServices/tests/Ancillary.Interop/SpanMarshallers.cs) is currently in the test assembly and I can't easily use it in System.Private.CoreLib?

Reply:
It is in a proposed API assembly that will hopefully become part of the official API surface. One could copy it over and use right now&mdash;see [`ArrayMarshaller`](https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/System/Runtime/InteropServices/ArrayMarshaller.cs)&mdash;but ideally the source generator would just use it and not require user intervention.

If you want to consume it now then it can be done just requires some effort. Do we think the current scenario is hot enough to warrant the effort?

Reply:
@AaronRobinsonMSFT It can wait until it becomes part of the official API surface, thank you!

BTW I love the source-generated based marshalling as I finally know what is happening behind the scenes ;)